deals-summary/src/app/extensions/DealsSummary.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Alert,
+  LoadingSpinner,
+  Statistics,
+  StatisticsItem,
+  Text,
+} from '@hubspot/ui-extensions';
+import { hubspot } from '@hubspot/ui-extensions';
+
+const DealsSummary = ({ runServerless }) => {
+  const [loading, setLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [dealsCount, setDealsCount] = useState(0);
+  const [totalAmount, setTotalAmount] = useState(0);
+
+  useEffect(() => {
+    const executeServerless = async () => {
+      const serverlessResponse = await runServerless({
+        name: 'get-data',
+        propertiesToSend: ['hs_object_id'],
+      });
+      setLoading(false);
+      if (serverlessResponse.status == 'SUCCESS') {
+        const { response } = serverlessResponse;
+        setDealsCount(response.dealsCount);
+        setTotalAmount(response.totalAmount);
+      } else {
+        setErrorMessage(serverlessResponse.message);
+      }
+    };
+    executeServerless();
```

Comment:
I think you should be able to avoid this superfluous async wrapper by just using `.then` instead of `await`ing `runServerless`.

Reply:
Also, if you rewrite this using `.then` chaining, you can more properly handle the error and loading states. For example if `runServerless` fails, the loading spinner will never turn off. Maybe something like this?

```jsx
runServerless({
  name: 'get-data',
  propertiesToSend: ['hs_object_id'],
}).then(serverlessResponse => {
  if (serverlessResponse.status == 'SUCCESS') {
    const { response } = serverlessResponse;
    setDealsCount(response.dealsCount);
    setTotalAmount(response.totalAmount);
  } else {
    setErrorMessage(serverlessResponse.message);
  }
}).finally(() => {
  setLoading(false);
});

```